### PR TITLE
story 1

### PR DESCRIPTION
**Pull request by lucyferrabee**

### CODE REVIEW

File: functions.php [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<?php
+
+function getDB(): PDO {
+$db = new PDO('mysql:host=db; dbname=collectionapp', 'root', 'password');
+
```
Comment (AlicjaZub) [inline functions.php]: empty line

File: functions.php [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+$db->setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_ASSOC);
+return $db;
+}
+
+function formatRecordsForDisplay(array $record)
```
Comment (AlicjaZub) [inline functions.php]: type of return?

File: functions.php [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+{
+    return 'Artist: ' . $record['Artist'] . '<br>' . 'Title:  ' . $record['Title'] . '<br>Rating:  ' .
+        $record['Rating'] . '<br>Release Date:  ' . $record['Released'] .
+        '<br>Condition:  ' . $record['Condition'] . '<br>' . '<br>';
+}
+
+function displayRecords(array $records) {
+    foreach($records as $record) {
+        echo formatRecordsForDisplay($record);
```
Comment (AlicjaZub) [inline functions.php:20]: if you don't want to have an echo statement inside a function, you can create a variable, put the output in the variable and return it outside the foreach loop -> to make it easier for unit testing

File: test/functions.php [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+use PHPUnit\Framework\TestCase;
+
+class functions extends TestCase{
+    public function testformatRecordsForDisplay()
+    {
+          $input = ["Artist"=>"artist","Title"=> "title", "Rating"=>"rating", "Released"=>1993, "Condition"=>"condition"];
+        $result = formatRecordsForDisplay($input);
+          $this->assertIsString($result);
+    }
+
```
Comment (AlicjaZub) [inline test/functions.php]: empty lines

File: test/functions.php [hunk 5]
```diff
@@ -0,0 +1,18 @@
+<?php
+
+require '../functions.php';
+
+use PHPUnit\Framework\TestCase;
+
+class functions extends TestCase{
+    public function testformatRecordsForDisplay()
```
Comment (AlicjaZub) [inline test/functions.php]: what type of test is it? success, failure etc?

File: test/functions.php [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+    {
+          $input = ["Artist"=>"artist","Title"=> "title", "Rating"=>"rating", "Released"=>1993, "Condition"=>"condition"];
+        $result = formatRecordsForDisplay($input);
+          $this->assertIsString($result);
+    }
+
+
+   }
+
+
```
Comment (AlicjaZub) [inline test/functions.php:23]: Would you add any more tests? For other functions for example or for the one you did, but like different types?

File: functions.php [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+}
+
+function formatRecordsForDisplay(array $record)
+{
+    return 'Artist: ' . $record['Artist'] . '<br>' . 'Title:  ' . $record['Title'] . '<br>Rating:  ' .
+        $record['Rating'] . '<br>Release Date:  ' . $record['Released'] .
+        '<br>Condition:  ' . $record['Condition'] . '<br>' . '<br>';
+}
+
+function displayRecords(array $records) {
```
Comment (AlicjaZub) [inline functions.php]: type of return?

File: functions.php [hunk 8]
```diff
@@ -0,0 +1,25 @@
+<?php
+
+function getDB(): PDO {
```
Comment (AlicjaZub) [inline functions.php]: you might put the { on a separate line (Mike/Neal/Lucia might have a problem with that, but I am not sure)

File: functions.php [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+function formatRecordsForDisplay(array $record): string
+{
+    return 'Artist: ' . $record['Artist'] . '<br>' . 'Title:  ' . $record['Title'] . '<br>Rating:  ' .
+       $record['Rating'] . '<br>Release Date:  ' . $record['Released'] .
+       '<br>Condition:  ' . $record['Condition'] . '<br>' . '<br>';
+}
+
+function displayRecords(array $records): string
+{
+foreach ($records as $record) {
```
Comment (AlicjaZub) [inline functions.php]: indentations for this whole function are gone

File: functions.php [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+echo formatRecordsForDisplay($record);
+}
+}
+
+
```
Comment (AlicjaZub) [inline functions.php]: empty lines

File: index.php [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+require 'functions.php';
+
+echo '<h1>Record Collection</h1>';
+echo '<br>';
+echo '<br>';
+
+$db = getDB();
+$query = $db->prepare('SELECT `Title`, `Rating`, `Released`, `Condition`, `Artist` FROM `records`;');
+$query->execute();
+
```
Comment (AlicjaZub) [inline index.php]: empty line

File: test/functions.php [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+require '../functions.php';
+
+use PHPUnit\Framework\TestCase;
+
+class functions extends TestCase
+{
+    public function testformatRecordsForDisplaySuccess()
+    {
+        $input = ["Artist"=>"artist","Title"=> "title", "Rating"=>"rating", "Released"=>1993, "Condition"=>"condition"];
```
Comment (AlicjaZub) [inline test/functions.php]: you can use single quotes - I don't know if it's something that the guys would flag <shrug>

also make sure that there are spaces before and after =>

File: functions.php (hunk 8 above)
Comment (nealioa) [reply functions.php]: Further to Alicja's comment, yes, the `{` goes on its own line when it's the opening one of a function.
And the following lines should be indented.

**Review by nealioa — CHANGES_REQUESTED** [review]

**Review by nealioa — APPROVED** [review]

**Pull request merged**